addons/addon-base-raas/packages/base-raas-services/lib/environment/__tests__/environment-mount-service.test.js

```diff
@@ -0,0 +1,213 @@
+    container.register('jsonSchemaValidationService', new JsonSchemaValidationService());
+    container.register('environmentScService', new EnvironmentScServiceMock());
+    container.register('environmentMountService', new EnvironmentMountService());
+    container.register('studyService', new StudyServiceMock());
+    container.register('studyPermissionService', new StudyPermissionServiceMock());
+    container.register('lockService', new LockServiceMock());
+    container.register('aws', new AwsServiceMock());
+    container.register('iamService', new IamServiceMock());
+    container.register('settings', new SettingsServiceMock());
+
+    await container.initServices();
+
+    // Get instance of the service we are testing
+    service = await container.find('environmentMountService');
+    environmentScService = await container.find('environmentScService');
+    iamService = await container.find('iamService');
+  });
+
+  describe('Update paths', () => {
+    it('should call nothing if all are admin changes', async () => {
+      // BUILD
+      const updateRequest = {
+        usersToAdd: [{ uid: 'User1-UID', permissionLevel: 'admin' }],
+        usersToRemove: [{ uid: 'User2-UID', permissionLevel: 'admin' }],
+      };
+      const studyId = 'StudyA';
+      service.addPermissions = jest.fn();
+      service.removePermissions = jest.fn();
+      service.updatePermissions = jest.fn();
+
+      // OPERATE
+      await service.applyWorkspacePermissions(studyId, updateRequest);
+
+      // CHECK
+      expect(service.addPermissions).not.toHaveBeenCalled();
+      expect(service.removePermissions).not.toHaveBeenCalled();
+      expect(service.updatePermissions).not.toHaveBeenCalled();
+    });
+
+    it('should call add when only adds are requested', async () => {
+      // BUILD
+      const updateRequest = {
+        usersToAdd: [{ uid: 'User1-UID', permissionLevel: 'readonly' }],
+        usersToRemove: [{ uid: 'User2-UID', permissionLevel: 'admin' }],
+      };
+      const studyId = 'StudyA';
+      service.addPermissions = jest.fn();
+      service.removePermissions = jest.fn();
+      service.updatePermissions = jest.fn();
+
+      // OPERATE
+      await service.applyWorkspacePermissions(studyId, updateRequest);
+
+      // CHECK
+      expect(service.addPermissions).toHaveBeenCalledWith([{ uid: 'User1-UID', permissionLevel: 'readonly' }], studyId);
+      expect(service.removePermissions).not.toHaveBeenCalled();
+      expect(service.updatePermissions).not.toHaveBeenCalled();
+    });
+
+    it('should call remove when only removals are requested', async () => {
+      // BUILD
+      const updateRequest = {
+        usersToAdd: [{ uid: 'User1-UID', permissionLevel: 'admin' }],
+        usersToRemove: [{ uid: 'User2-UID', permissionLevel: 'readwrite' }],
+      };
+      const studyId = 'StudyA';
+      service.addPermissions = jest.fn();
+      service.removePermissions = jest.fn();
+      service.updatePermissions = jest.fn();
```

Comment:
Can we add tests for these methods- add, remove and update or not mock them out in some of the tests. Not sure if we are covering these tests in unit tests.

Reply:
I believe those would be prime candidates for integration tests.

Reply:
As discussed, I'll add JSON schema validation.

Reply:
As discussed, pushed more unit tests that ensure logic is intact.